src/utils/builder.ts

```diff
@@ -329,25 +275,70 @@ export function setTransitionCondition(
   startNodeKey: string,
   transitionId: string,
   description: string,
-  elm: ElmLibrary
-): void {
+  elm: ElmLibrary,
+  criteria?: string
```

Comment:
I'm getting confused throwing criteria around - this refers to the criteria label right? Naming it `criteriaLabel` would be clearer since its not the entire criteria object.